src/librustdoc/clean/mod.rs

```diff
@@ -421,8 +430,12 @@ impl Clean<GenericParamDef> for ty::GenericParamDef {
                     // the cleaning process of the type itself. To resolve this and have the
                     // `self_def_id` set, we override it here.
                     // See https://github.com/rust-lang/rust/issues/85454
-                    if let QPath { ref mut self_def_id, .. } = default {
-                        *self_def_id = cx.tcx.parent(self.def_id);
+                    if let QPath { ref mut should_show_cast, ref trait_, ref self_type, .. } =
+                        default
+                    {
+                        let self_def_id = cx.tcx.parent(self.def_id);
+                        *should_show_cast =
+                            compute_should_show_cast(self_def_id, trait_, self_type);
                     }
```

Comment:
The main reason I wanted to get rid of the field was to eliminate this confusing mutation of it. Can you see if there's a way to figure this out ahead-of-time and get rid of this mutation of `should_show_cast`?

Reply:
Worth a try!